src/perl/lib/wtsi_clarity/util/csv/factory.pm

```diff
@@ -27,6 +27,8 @@ sub create {
     $csv_factory = wtsi_clarity::util::csv::factories::generic_csv_reader->new();
   } elsif ( $csv_type eq 'calliper_reader') {
     $csv_factory = wtsi_clarity::util::csv::factories::calliper_csv_reader->new();
+  } elsif ( $csv_type eq 'fluidigm_writer') {
+    $csv_factory = wtsi_clarity::util::csv::factories::generic_csv_writer->new();
```

Comment:
It is getting very ugly.
If we would have time I would refactor it using the Strategy Pattern: https://en.wikipedia.org/wiki/Strategy_pattern